app/controllers/course/registrations_controller.rb

```diff
@@ -0,0 +1,158 @@
+      return
+    end
+    @registration.user = current_user
+    @registration.save!
+
+    # TODO: Notify lecturer
+    respond_to do |format|
+      format.html do
+        redirect_to course_path(@course), flash: { notice: t('.request_format') %
+          { role: t("course.registrations.role.#{role}") } }
+      end
+    end
+  end
+
+  def update #:nodoc:
+    if params[:update_single]
+      update_registration
+    elsif params[:approve_single]
+      approve_registration
+    elsif params[:delete_single]
+      redirect_to delete_course_registration_path(@course, @registration)
+    else
+      redirect_to course_path(@course), status: :bad_request
+    end
+  end
+
+  def delete #:nodoc:
+    message = t('.notice_format') % { role: t("course.registrations.role.#{@registration.role}"),
+                                      user: @registration.user.name }
+    @registration.destroy!
+
+    respond_to do |format|
+      format.html do
+        redirect_to course_registrations_path(@course), flash: { notice: message }
+      end
+    end
+  end
+
+  def update_multiple #:nodoc:
+    if should_process_single?
+      redirect_to update_course_registration_path(@course, single_registration_id, params)
+      return
+    end
+
+    flash = {}
+    begin
+      processed_ids = Course::Registration.process!(@course,
+                                                    params[:registration_ids],
+                                                    approve: should_approve?,
+                                                    process_all: should_process_all?,
+                                                    role: update_scope)
+      flash[:notice] = (should_approve?) ? t('.approve_notice_format') : t('.delete_notice_format')
+      flash[:notice] %= { count: processed_ids.count }
+    rescue
+      flash[:error] = t('.error_format') % { reason: $!.message }
+    end
+
+    respond_to do |format|
+      format.html { redirect_to course_registrations_path(@course), flash: flash }
+    end
+  end
+
+  private
+
+  def should_process_single? #:nodoc:
+    params[:approve_single] || params[:delete_single]
+  end
+
+  def single_registration_id
+    result = params[:approve_single]
+    result ||= params[:delete_single]
+  end
+
+  def should_process_all? #:nodoc:
+    params[:approve_all] || params[:delete_all]
+  end
+
+  def should_approve? #:nodoc:
+    params[:approve_single] || params[:approve_selected] || params[:approve_all]
+  end
+
+  def update_scope
+    result = params[:approve_all]
+    result ||= params[:delete_all]
+  end
+
+  def update_registration #:nodoc:
+    @registration.update_attributes(registration_params)
+
+    respond_to do |format|
+      format.html do
+        redirect_to course_path(@course),
+                    flash: { notice: t('.update_notice_format') %
+                      { role: t("course.registrations.role.#{@registration.role}") } }
+      end
+    end
+  end
+
+  def approve_registration #:nodoc:
```

Comment:
Assignment Branch Condition size for approve_registration is too high. [16.06/15]<br>Method has too many lines. [14/10]